File: src/foundation/java/io/karma/pda/foundation/component/RecipeImage.java
/*
 * Copyright (C) 2024 Karma Krafts & associates
 */

package io.karma.pda.foundation.component;

import io.karma.pda.api.app.component.AbstractComponent;
import io.karma.pda.api.app.component.ComponentType;
import io.karma.pda.api.state.MutableState;
import io.karma.pda.api.state.Synchronize;
import io.karma.peregrine.api.color.Color;
import io.karma.peregrine.api.color.ColorProvider;
import net.minecraft.resources.ResourceLocation;

import java.util.UUID;

/**
 * @author Alexander Hinze
 * @since 13/04/2024
 */
public class RecipeImage extends AbstractComponent {
    @Synchronize
    public final MutableState<ResourceLocation> recipe = MutableState.of(new ResourceLocation("bucket"));
    @Synchronize
    public final MutableState<ColorProvider> background = MutableState.of(Color.NONE);
    @Synchronize
    public final MutableState<ColorProvider> foreground = MutableState.of(Color.NONE);

    public RecipeImage(final ComponentType<?> type, final UUID id) {
        super(type, id);
    }
}

File: src/composition/kotlin/io/karma/pda/composition/component/RecipeImage.kt
/*
 * Copyright (C) 2024 Karma Krafts & associates
 */

@file:JvmName("ComponentExtensionsKt")
@file:JvmMultifileClass

package io.karma.pda.composition.component

import io.karma.pda.api.app.component.Container
import io.karma.pda.composition.Composable
import io.karma.pda.composition.Composer
import io.karma.pda.composition.LayoutProps
import io.karma.pda.foundation.component.DefaultComponents
import io.karma.pda.foundation.component.RecipeImage

/**
 * @author Alexander Hinze
 * @since 07/06/2024
 */

@Composable
inline fun <C : Container> Composer<C>.RecipeImage(
    crossinline layoutProps: LayoutProps,
    localName: String? = null,
    props: @Composable RecipeImage.() -> Unit = {}
) = component(
    type = DefaultComponents.RECIPE_IMAGE,
    layoutProps = layoutProps,
    localName = localName,
    props = props
)